@MartinHjelmare running into this when trying to test:
```
        # Test turn on setting speed
        await hass.services.async_call(
            "fan",
            "turn_on",
            {"entity_id": FAN_ENTITY, "speed" : "medium"},
            blocking=True,
        )

>       assert len(client.async_send_command.call_args_list) == 1
E       assert 0 == 1
E         +0
E         -1
```
Any ideas?

--- a/homeassistant/components/zwave_js/fan.py
+++ b/homeassistant/components/zwave_js/fan.py
@@ -67,8 +67,7 @@ class ZwaveFan(ZWaveBaseEntity, FanEntity):
     async def async_set_speed(self, speed: Any) -> None:
         """Set the speed of the fan."""
         if speed not in SPEED_TO_VALUE:
-            _LOGGER.warning("Invalid speed received: %s", speed)
-            return
+            raise ValueError(f"Invalid speed received: {speed}")
         self._previous_speed = speed
         target_value = self.get_zwave_value("targetValue")
         self.info.node.async_set_value(target_value, SPEED_TO_VALUE[speed])

--- a/homeassistant/components/zwave_js/fan.py
+++ b/homeassistant/components/zwave_js/fan.py
@@ -70,21 +70,21 @@ class ZwaveFan(ZWaveBaseEntity, FanEntity):
             raise ValueError(f"Invalid speed received: {speed}")
         self._previous_speed = speed
         target_value = self.get_zwave_value("targetValue")
-        self.info.node.async_set_value(target_value, SPEED_TO_VALUE[speed])
+        await self.info.node.async_set_value(target_value, SPEED_TO_VALUE[speed])
 
     async def async_turn_on(self, speed: Optional[Any] = None, **kwargs: Any) -> None:
         """Turn the device on."""
         if speed is None:
             # Value 255 tells device to return to previous value
             target_value = self.get_zwave_value("targetValue")
-            self.info.node.async_set_value(target_value, 255)
+            await self.info.node.async_set_value(target_value, 255)
         else:
             await self.async_set_speed(speed)
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the device off."""
         target_value = self.get_zwave_value("targetValue")
-        self.info.node.async_set_value(target_value, 0)
+        await self.info.node.async_set_value(target_value, 0)
 
     @property
     def is_on(self) -> bool:

--- a/homeassistant/components/zwave_js/fan.py
+++ b/homeassistant/components/zwave_js/fan.py
@@ -62,9 +62,9 @@ class ZwaveFan(ZWaveBaseEntity, FanEntity):
     ) -> None:
         """Initialize the fan."""
         super().__init__(config_entry, client, info)
-        self._previous_speed = None
+        self._previous_speed: Optional[str] = None
 
-    async def async_set_speed(self, speed: Any) -> None:
+    async def async_set_speed(self, speed: str) -> None:
         """Set the speed of the fan."""
         if speed not in SPEED_TO_VALUE:
             raise ValueError(f"Invalid speed received: {speed}")
@@ -72,7 +72,7 @@ class ZwaveFan(ZWaveBaseEntity, FanEntity):
         target_value = self.get_zwave_value("targetValue")
         await self.info.node.async_set_value(target_value, SPEED_TO_VALUE[speed])
 
-    async def async_turn_on(self, speed: Optional[Any] = None, **kwargs: Any) -> None:
+    async def async_turn_on(self, speed: Optional[str] = None, **kwargs: Any) -> None:
         """Turn the device on."""
         if speed is None:
             # Value 255 tells device to return to previous value

--- a/tests/components/zwave_js/test_fan.py
+++ b/tests/components/zwave_js/test_fan.py
@@ -1,4 +1,5 @@
 """Test the Z-Wave JS fan platform."""
+import pytest
 from zwave_js_server.event import Event
 
 from homeassistant.components.fan import ATTR_SPEED, SPEED_MEDIUM
@@ -46,11 +47,22 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
 
     client.async_send_command.reset_mock()
 
-    # Test setting speed
+    # Test setting unknown speed
+    with pytest.raises(ValueError):
+        await hass.services.async_call(
+            "fan",
+            "set_speed",
+            {"entity_id": FAN_ENTITY, "speed": 99},
+            blocking=True,
+        )
+
+    client.async_send_command.reset_mock()
+
+    # Test turn on no speed
     await hass.services.async_call(
         "fan",
-        "set_speed",
-        {"entity_id": FAN_ENTITY, "speed": 99},
+        "turn_on",
+        {"entity_id": FAN_ENTITY},
         blocking=True,
     )
 
@@ -74,7 +86,7 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
             "label": "Target value",
         },
     }
-    assert args["value"] == 99
+    assert args["value"] == 255
 
     client.async_send_command.reset_mock()
 

--- a/tests/components/zwave_js/test_fan.py
+++ b/tests/components/zwave_js/test_fan.py
@@ -1,6 +1,8 @@
 """Test the Z-Wave JS fan platform."""
 from zwave_js_server.event import Event
 
+from homeassistant.components.fan import ATTR_SPEED, SPEED_MEDIUM
+
 FAN_ENTITY = "fan.in_wall_smart_fan_control_current_value"
 
 
@@ -16,7 +18,7 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
     await hass.services.async_call(
         "fan",
         "turn_on",
-        {"entity_id": FAN_ENTITY, "speed": "medium"},
+        {"entity_id": FAN_ENTITY, "speed": SPEED_MEDIUM},
         blocking=True,
     )
 
@@ -44,50 +46,18 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
 
     client.async_send_command.reset_mock()
 
-    # Test setting position
+    # Test setting speed
     await hass.services.async_call(
-        "cover",
-        "set_cover_position",
-        {"entity_id": FAN_ENTITY, "position": 0},
+        "fan",
+        "set_speed",
+        {"entity_id": FAN_ENTITY, "speed": 99},
         blocking=True,
     )
 
     assert len(client.async_send_command.call_args_list) == 1
     args = client.async_send_command.call_args[0][0]
     assert args["command"] == "node.set_value"
-    assert args["nodeId"] == 6
-    assert args["valueId"] == {
-        "commandClassName": "Multilevel Switch",
-        "commandClass": 38,
-        "endpoint": 0,
-        "property": "targetValue",
-        "propertyName": "targetValue",
-        "metadata": {
-            "label": "Target value",
-            "max": 99,
-            "min": 0,
-            "type": "number",
-            "readable": True,
-            "writeable": True,
-            "label": "Target value",
-        },
-    }
-    assert args["value"] == 0
-
-    client.async_send_command.reset_mock()
-
-    # Test opening
-    await hass.services.async_call(
-        "cover",
-        "open_cover",
-        {"entity_id": FAN_ENTITY},
-        blocking=True,
-    )
-
-    assert len(client.async_send_command.call_args_list) == 1
-    args = client.async_send_command.call_args[0][0]
-    assert args["command"] == "node.set_value"
-    assert args["nodeId"] == 6
+    assert args["nodeId"] == 17
     assert args["valueId"] == {
         "commandClassName": "Multilevel Switch",
         "commandClass": 38,
@@ -108,33 +78,10 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
 
     client.async_send_command.reset_mock()
 
-    # Test position update from value updated event
-    event = Event(
-        type="value updated",
-        data={
-            "source": "node",
-            "event": "value updated",
-            "nodeId": 6,
-            "args": {
-                "commandClassName": "Multilevel Switch",
-                "commandClass": 38,
-                "endpoint": 0,
-                "property": "currentValue",
-                "newValue": 99,
-                "prevValue": 0,
-                "propertyName": "currentValue",
-            },
-        },
-    )
-    node.receive_event(event)
-
-    state = hass.states.get(FAN_ENTITY)
-    assert state.state == "open"
-
-    # Test closing
+    # Test turning off
     await hass.services.async_call(
-        "cover",
-        "close_cover",
+        "fan",
+        "turn_off",
         {"entity_id": FAN_ENTITY},
         blocking=True,
     )
@@ -142,7 +89,7 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
     assert len(client.async_send_command.call_args_list) == 1
     args = client.async_send_command.call_args[0][0]
     assert args["command"] == "node.set_value"
-    assert args["nodeId"] == 6
+    assert args["nodeId"] == 17
     assert args["valueId"] == {
         "commandClassName": "Multilevel Switch",
         "commandClass": 38,
@@ -163,12 +110,38 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
 
     client.async_send_command.reset_mock()
 
+    # Test speed update from value updated event
     event = Event(
         type="value updated",
         data={
             "source": "node",
             "event": "value updated",
-            "nodeId": 6,
+            "nodeId": 17,
+            "args": {
+                "commandClassName": "Multilevel Switch",
+                "commandClass": 38,
+                "endpoint": 0,
+                "property": "currentValue",
+                "newValue": 99,
+                "prevValue": 0,
+                "propertyName": "currentValue",
+            },
+        },
+    )
+    node.receive_event(event)
+
+    state = hass.states.get(FAN_ENTITY)
+    assert state.state == "on"
+    assert state.attributes[ATTR_SPEED] == "high"
+
+    client.async_send_command.reset_mock()
+
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 17,
             "args": {
                 "commandClassName": "Multilevel Switch",
                 "commandClass": 38,
@@ -183,4 +156,5 @@ async def test_fan(hass, client, in_wall_smart_fan_control, integration):
     node.receive_event(event)
 
     state = hass.states.get(FAN_ENTITY)
-    assert state.state == "closed"
+    assert state.state == "off"
+    assert state.attributes[ATTR_SPEED] == "off"

